runtime/src/message_processor.rs

```diff
@@ -58,7 +58,8 @@ impl MessageProcessor {
         transaction_context: &mut TransactionContext,
         rent: Rent,
         log_collector: Option<Rc<RefCell<LogCollector>>>,
-        tx_executor_cache: Rc<RefCell<TransactionExecutorCache>>,
+        programs_loaded_for_tx_batch: Rc<RefCell<LoadedProgramsForTxBatch>>,
```

Comment:
These two might not even require `Rc<RefCell<>>`, but that can be refactored later.